lib/server/routes/frames.js

```diff
@@ -244,6 +246,10 @@ FramesRouter.prototype.addShardToFrame = function(req, res, next) {
     ));
   }
 
+  if (req.body.size > MAX_SHARD_SIZE) {
+    return next(new errors.BadRequestError('Maximum shard size'));
```

Comment:
change to `return next(new errors.BadRequestError('Shard of size ' + req.body.size + ' is larger than maximum shard size:  ' + MAX_SHARD_SIZE));`